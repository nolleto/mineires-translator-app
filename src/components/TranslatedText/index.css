.translated-text {
  display: flex;
  flex-direction: column;
  align-content: center;
  text-align: start;
  margin: 40px 10px 10px 10px;
}

.translated-text__span {
  font-size: 18px;
  font-weight: bold;
}
