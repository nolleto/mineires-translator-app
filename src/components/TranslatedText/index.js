import './index.css';

import React from 'react';
import TranslatedLines from '../TranslatedLines';

const TranslatedText = ({ text }) => {
  return (
    <div className='translated-text'>
      <span className='translated-text__span'>
        Texto em MineirÃªs
      </span>

      <TranslatedLines
        lines={text.split('\n')}
      />
    </div>
  )
}

export default TranslatedText
