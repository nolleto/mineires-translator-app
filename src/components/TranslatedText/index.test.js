import React from 'react';
import TranslatedText from './index';
import { shallow } from 'enzyme';

const shallowComponent = (props = {}) => {
  const finalProps = {
    text: '',
    ...props
  };

  return shallow(
    <TranslatedText
      {...finalProps}
    />
  );
};

describe('TranslatedText component', () => {
  describe('Snapshot', () => {
    it('when has no text', () => {
      const snapshot = shallowComponent({ text: '' });

      expect(snapshot).toMatchSnapshot();
    });

    it('when has text', () => {
      const snapshot = shallowComponent({
        text: 'some text'
      });

      expect(snapshot).toMatchSnapshot();
    });

    it('when has two text', () => {
      const snapshot = shallowComponent({
        text: 'some text\nanother text'
      });

      expect(snapshot).toMatchSnapshot();
    });
  });
});
