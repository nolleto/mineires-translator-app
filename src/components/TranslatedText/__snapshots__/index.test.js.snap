// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TranslatedText component Snapshot when has no text 1`] = `
<div
  className="translated-text"
>
  <span
    className="translated-text__span"
  >
    Texto em Mineirês
  </span>
  <TranslatedLines
    lines={
      Array [
        "",
      ]
    }
  />
</div>
`;

exports[`TranslatedText component Snapshot when has text 1`] = `
<div
  className="translated-text"
>
  <span
    className="translated-text__span"
  >
    Texto em Mineirês
  </span>
  <TranslatedLines
    lines={
      Array [
        "some text",
      ]
    }
  />
</div>
`;

exports[`TranslatedText component Snapshot when has two text 1`] = `
<div
  className="translated-text"
>
  <span
    className="translated-text__span"
  >
    Texto em Mineirês
  </span>
  <TranslatedLines
    lines={
      Array [
        "some text",
        "another text",
      ]
    }
  />
</div>
`;
